indico/core/db/sqlalchemy/protection.py

```diff
@@ -212,10 +212,10 @@ def can_access(self, user, allow_admin=True):
                 # mixin or a legacy object with an AccessController
                 parent = self.protection_parent
                 if parent is None:
-                    # This should be the case for the top-level object,
-                    # i.e. the root category, which shouldn't allow
-                    # ProtectionMode.inheriting as it makes no sense.
-                    raise TypeError(f'protection_parent of {self} is None')
```

Comment:
Can we be more explicit, something like `allow_none_protection_parent`? I'd rather not disable this exception anywhere but on `Event`